velox/common/encode/Base64.cpp

```diff
@@ -324,65 +329,62 @@ uint8_t Base64::Base64ReverseLookup(
 
 size_t
 Base64::decode(const char* src, size_t src_len, char* dst, size_t dst_len) {
-  return decodeImpl(src, src_len, dst, dst_len, kBase64ReverseIndexTable, true);
+  return decodeImpl(src, src_len, dst, dst_len, kBase64ReverseIndexTable);
 }
 
 // static
-size_t
-Base64::calculateDecodedSize(const char* data, size_t& size, bool withPadding) {
+size_t Base64::calculateDecodedSize(const char* data, size_t& size) {
   if (size == 0) {
     return 0;
   }
 
-  auto needed = (size / 4) * 3;
-  if (withPadding) {
-    // If the pad characters are included then the source string must be a
-    // multiple of 4 and we can query the end of the string to see how much
-    // padding exists.
-    if (size % 4 != 0) {
+  // Check if the input data is padded
+  if (isPadded(data, size)) {
```

Comment:
isPadded simply checks that the last character is '='. However, it is possible that proper padding requires two = chars, but there is only one. Do we want to throw in this case? If so, it would be nice to document that clearly. 

BTW, since we are changing the semantics of a Presto function, let's update its documentation: https://facebookincubator.github.io/velox/functions/presto/binary.html#from_base64

Reply:
Yes, `isPadded` checks whether any pads are at the end of the encoded string. We don't throw any errors based on the count of pads, but a throw will happen if the input string is not a multiple of the proper block size (4 in the case of base64).

In future PRs, `isPadded` will be moved to a utility class that base64 and base32 will use (https://github.com/facebookincubator/velox/pull/8651 - This is how base64.h/cpp will look once all my planned changes are incorporated).

We can go ahead with either of two ways,
1. Implement a common utility class that base32 and base64 will use - https://github.com/facebookincubator/velox/pull/8651/
2. Keep base32 and base64 separate and improvise individual logic with implementations specific to its respective base type.

Please share your thoughts on it. We were thinking of choosing the first route and have implemented it.